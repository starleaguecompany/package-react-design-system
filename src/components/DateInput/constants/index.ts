export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
