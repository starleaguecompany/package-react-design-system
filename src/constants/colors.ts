export const COLORS = {
  GRAY: 'gray',
  GREEN: 'green',
  BLUE: 'blue',
  ORANGE: 'orange',
  VIOLET: 'violet',
  RED: 'red',
} as const
