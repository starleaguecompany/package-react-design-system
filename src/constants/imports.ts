export const reactDSImportPath = '@starleaguecompany/react-design-system';
