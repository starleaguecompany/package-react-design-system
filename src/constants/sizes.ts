export const CONTENT_SIZES = {
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
} as const;

export const CONTAINER_SIZES = {
  S36: 36,
  S44: 44,
  S52: 52,
  S60: 60,
} as const;

export const FONT_SIZES = {
  S10: 10,
  S12: 12,
  S14: 14,
  S16: 16,
  S20: 20,
} as const;
