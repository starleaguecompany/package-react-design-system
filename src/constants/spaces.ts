export const SPACE_SIZES = {
  S0: 0,
  S2: 2,
  S4: 4,
  S8: 8,
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
  S32: 32,
  S36: 36,
  S40: 40,
} as const;
