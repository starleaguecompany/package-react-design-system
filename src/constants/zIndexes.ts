export const Z_INDEXES = {
  DROPDOWN: 1060,
  STICKY: 1020,
  FIXED: 1030,
  MODAL_BACKDROP: 1040,
  MODAL: 1050,
  TOOLTIP: 1060,
} as const;
