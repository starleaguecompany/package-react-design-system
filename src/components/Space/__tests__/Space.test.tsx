import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { SPACE_SIZES } from '../../../constants/spaces';
import { Space, Button } from '../../../index';

describe('Space', () => {
  test('should render correctly with no props', () => {
    render(<Space />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveClass('container direction-horizontal', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<Space id="test-id" className="test-class" />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveAttribute('id', 'test-id');
    expect(screen.getByTestId('Space')).toHaveClass('test-class');
  });

  test('should render correctly with direction prop', () => {
    render(<Space direction="vertical" />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveClass('direction-vertical');
    expect(screen.getByTestId('Space')).not.toHaveClass('direction-horizontal');
  });

  test('should render correctly with size prop', () => {
    render(<Space size={SPACE_SIZES.S24} />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveStyle('columnGap: 24px');
    expect(screen.getByTestId('Space')).toHaveStyle('rowGap: 24px');
  });

  test('should render correctly with align prop', () => {
    render(<Space align="end" />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveClass('align-end');
  });

  test('should render correctly with justify prop', () => {
    render(<Space justify="space-between" />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveClass('justify-space-between');
  });

  test('should render correctly with inline prop', () => {
    render(<Space inline />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveClass('inline');
  });

  test('should render correctly with wrap prop', () => {
    render(<Space wrap />);

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveClass('wrap');
  });

  test('should render correctly with children', () => {
    render(
      <Space>
        <Button>1</Button>
        <Button>2</Button>
      </Space>
    );

    expect(screen.getByTestId('Space')).toMatchSnapshot();
    expect(screen.getByTestId('Space')).toHaveTextContent('12');
    expect(screen.getByRole('button', { name: '1' })).toBeVisible();
    expect(screen.getByRole('button', { name: '2' })).toBeVisible();
  });
});
