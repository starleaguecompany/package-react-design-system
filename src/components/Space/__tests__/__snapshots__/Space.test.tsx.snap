// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space should render correctly with align prop 1`] = `
<div
  class="container direction-horizontal align-end"
  data-qa="Space"
/>
`;

exports[`Space should render correctly with attributes 1`] = `
<div
  class="test-class container direction-horizontal"
  data-qa="Space"
  id="test-id"
/>
`;

exports[`Space should render correctly with children 1`] = `
<div
  class="container direction-horizontal"
  data-qa="Space"
>
  <button
    class="container size-44 color-gray-outlined"
    data-qa="Button"
  >
    <span
      class="text"
    >
      1
    </span>
  </button>
  <button
    class="container size-44 color-gray-outlined"
    data-qa="Button"
  >
    <span
      class="text"
    >
      2
    </span>
  </button>
</div>
`;

exports[`Space should render correctly with direction prop 1`] = `
<div
  class="container direction-vertical"
  data-qa="Space"
/>
`;

exports[`Space should render correctly with inline prop 1`] = `
<div
  class="container direction-horizontal inline"
  data-qa="Space"
/>
`;

exports[`Space should render correctly with justify prop 1`] = `
<div
  class="container direction-horizontal justify-space-between"
  data-qa="Space"
/>
`;

exports[`Space should render correctly with no props 1`] = `
<div
  class="container direction-horizontal"
  data-qa="Space"
/>
`;

exports[`Space should render correctly with size prop 1`] = `
<div
  class="container direction-horizontal"
  data-qa="Space"
  style="column-gap: 24px; row-gap: 24px;"
/>
`;

exports[`Space should render correctly with wrap prop 1`] = `
<div
  class="container direction-horizontal wrap"
  data-qa="Space"
/>
`;
